pkg/virt-controller/watch/migration.go

```diff
@@ -2,59 +2,92 @@ package watch
 
 import (
 	"fmt"
+	"time"
 
 	"k8s.io/client-go/kubernetes"
 	kubeapi "k8s.io/client-go/pkg/api"
 	k8sv1 "k8s.io/client-go/pkg/api/v1"
 	metav1 "k8s.io/client-go/pkg/apis/meta/v1"
 	"k8s.io/client-go/pkg/fields"
 	"k8s.io/client-go/pkg/labels"
-	"k8s.io/client-go/pkg/types"
+	"k8s.io/client-go/pkg/runtime"
+	"k8s.io/client-go/pkg/util/wait"
 	"k8s.io/client-go/pkg/util/workqueue"
 	"k8s.io/client-go/rest"
 	"k8s.io/client-go/tools/cache"
-	"k8s.io/client-go/tools/record"
-
 	kubev1 "kubevirt.io/kubevirt/pkg/api/v1"
 	"kubevirt.io/kubevirt/pkg/kubecli"
 	"kubevirt.io/kubevirt/pkg/logging"
 	"kubevirt.io/kubevirt/pkg/virt-controller/services"
 )
 
-func NewMigrationController(migrationService services.VMService, recorder record.EventRecorder, restClient *rest.RESTClient, clientset *kubernetes.Clientset) (cache.Store, *kubecli.Controller, *workqueue.RateLimitingInterface) {
+func NewMigrationController(migrationService services.VMService, restClient *rest.RESTClient, clientset *kubernetes.Clientset) *MigrationController {
 	lw := cache.NewListWatchFromClient(restClient, "migrations", k8sv1.NamespaceDefault, fields.Everything())
 	queue := workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter())
-	store, controller := kubecli.NewController(lw, queue, &kubev1.Migration{}, NewMigrationControllerDispatch(migrationService, restClient, clientset))
-	return store, controller, &queue
-}
-
-func NewMigrationControllerDispatch(vmService services.VMService, restClient *rest.RESTClient, clientset *kubernetes.Clientset) kubecli.ControllerDispatch {
-
-	dispatch := MigrationDispatch{
+	store, informer := cache.NewIndexerInformer(lw, &kubev1.Migration{}, 0, kubecli.NewResourceEventHandlerFuncsForWorkqueue(queue), cache.Indexers{})
+	return &MigrationController{
 		restClient: restClient,
-		vmService:  vmService,
+		vmService:  migrationService,
 		clientset:  clientset,
+		queue:      queue,
+		store:      store,
+		informer:   informer,
 	}
-	return &dispatch
 }
 
-type MigrationDispatch struct {
+type MigrationController struct {
 	restClient *rest.RESTClient
 	vmService  services.VMService
 	clientset  *kubernetes.Clientset
+	queue      workqueue.RateLimitingInterface
+	store      cache.Store
+	informer   cache.ControllerInterface
+}
+
+func (c *MigrationController) Run(threadiness int, stopCh chan struct{}) {
+	defer kubecli.HandlePanic()
+	defer c.queue.ShutDown()
+	logging.DefaultLogger().Info().Msg("Starting controller.")
+
+	// Start all informers and wait for the cache sync
+	_, jobInformer := NewMigrationJobInformer(c.clientset, c.queue)
```

Comment:
It is really confusing to call this a MigrationJob.  it Is not a K8S Job.  

There is the pod that runs the migration, and the pod that is the mirgation target.  Suggest that the first be called the MigrationProcessPod and the second the MigrationTargetPod